include/envoy/ssl/context_config.h

```diff
@@ -63,6 +63,16 @@ class ContextConfig {
    * Otherwise, ""
    */
   virtual const std::string& verifyCertificateHash() const PURE;
+
+  /**
+   * @return The minimum TLS protocol version to negotiate.
+   */
+  virtual unsigned minProtocolVersion() const PURE;
```

Comment:
nit: at interface level I would probably make this uint64_t, though I understand the boringssl data type is unsigned. NBD either way.

Reply:
BoringSSL uses `uint16_t` for versions, but I didn't want to leak it to the interface level and `unsigned` felt cleaner, but I don't really care which one we use.